### ISSUE TITLE
SSLContext infinite recursion in Python 3.6

### ISSUE THREAD

**Issue by juokaz** [gevent/gevent#903]
The issue is this method in `python/ssl.py`.

```
    @options.setter
    def options(self, value):
        super(SSLContext, SSLContext).options.__set__(self, value)
```

It fails when using for example `requests` library:

```
root@8042e3f57981:/app# python
Python 3.6.0b4 (default, Nov 23 2016, 21:34:29)
[GCC 4.9.2] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gevent.monkey
>>> gevent.monkey.patch_all()
>>>
>>> from requests.packages.urllib3.util.ssl_ import create_urllib3_context
>>> create_urllib3_context()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/util/ssl_.py", line 268, in create_urllib3_context
    context.options |= options
  File "/usr/local/lib/python3.6/ssl.py", line 459, in options
    super(SSLContext, SSLContext).options.__set__(self, value)
  File "/usr/local/lib/python3.6/ssl.py", line 459, in options
    super(SSLContext, SSLContext).options.__set__(self, value)
  File "/usr/local/lib/python3.6/ssl.py", line 459, in options
    super(SSLContext, SSLContext).options.__set__(self, value)
  [Previous line repeated 329 more times]
RecursionError: maximum recursion depth exceeded while calling a Python object
>>>
```

**Comment by jamadden** [comment]
I believe you're using gevent 1.1, yes? Please use gevent 1.2 on Python 3.6.

**Comment by juokaz** [comment]
I tried running on 1.2a1 and it worked. My bad, thank you @jamadden!

**Comment by ZekeLee** [comment]
My gevent version is 1.2.1 and still get this error...


`bogon:~ zekelee$ pip3 show gevent
Name: gevent
Version: 1.2.1
Summary: Coroutine-based network library
Home-page: http://www.gevent.org/
Author: Jason Madden
Author-email: jason@nextthought.com
License: MIT
Location: /usr/local/lib/python3.6/site-packages
Requires: greenlet
bogon:~ zekelee$ python3 -V
Python 3.6.0
bogon:~ zekelee$ python3 Desktop/ttt.py 
Traceback (most recent call last):
  File "Desktop/ttt.py", line 5, in <module>
    response = requests.get('https://www.baidu.com', timeout = 3600)
  File "/usr/local/lib/python3.6/site-packages/requests/api.py", line 70, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/requests/sessions.py", line 488, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.6/site-packages/requests/sessions.py", line 609, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/requests/adapters.py", line 423, in send
    timeout=timeout
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 594, in urlopen
    chunked=chunked)
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 350, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/connectionpool.py", line 835, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/connection.py", line 311, in connect
    cert_reqs=resolve_cert_reqs(self.cert_reqs),
  File "/usr/local/lib/python3.6/site-packages/requests/packages/urllib3/util/ssl_.py", line 264, in create_urllib3_context
    context.options |= options
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/ssl.py", line 459, in options
    super(SSLContext, SSLContext).options.__set__(self, value)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/ssl.py", line 459, in options
    super(SSLContext, SSLContext).options.__set__(self, value)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/ssl.py", line 459, in options
    super(SSLContext, SSLContext).options.__set__(self, value)
  [Previous line repeated 323 more times]
RecursionError: maximum recursion depth exceeded`

**Comment by pirate** [comment]
Now that python3.6 is released, should we re-open this ticket until gevent >1.2 reaches non-alpha and becomes default pip install?  As it stands the "stable" 3.6 standard library with a default setup is not usable with https urls.

**Comment by jamadden** [comment]
gevent 1.2 is released and works with 3.6.

**Comment by pirate** [comment]
Ah thanks @jamadden, it looks like I mistakenly thought `pip install --upgrade requests` would also bump `gevent` to the newer required version.  Working for me now after `pip install --upgrade gevent`.
